### PR TITLE
[DataGrid] Fix typos in the JSDoc

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

File: docs/translations/api-docs/data-grid/data-grid-premium.json [hunk 1]
```diff
@@ -942,7 +942,7 @@
       "typeDescriptions": {}
     },
     "throttleRowsMs": {
-      "description": "If positive, the Grid will throttle updates coming from <code>apiRef.current.updateRows</code> and <code>apiRef.current.setRows</code>. It can be useful if you have a high update rate but do not want to do heavy work like filtering / sorting or rendering on each  individual update.",
```
Comment (flaviendelangle) [inline docs/translations/api-docs/data-grid/data-grid-premium.json:945]: We always use `grid` in the other props JSDoc

Comment (LukasTy) [reply docs/translations/api-docs/data-grid/data-grid-premium.json:945]: Was going with Data Grid here ever considered? 🤔 
https://www.notion.so/mui-org/Style-conventions-for-MUI-components-and-products-dff7d2d7d1ba4a4abd47f48cf345b800

Comment (flaviendelangle) [reply docs/translations/api-docs/data-grid/data-grid-premium.json:945]: But I do agree `Data Grid` it probably the way to go

**Review by cherniavskii — APPROVED** [review]
Thanks!

**Comment by flaviendelangle** [comment]
Thanks for the follow up @cherniavskii !

**Pull request merged**